pkg/_pub_shared/lib/data/package_api.dart

```diff
@@ -85,17 +85,25 @@ class GithubPublishingConfig {
   String? tagPattern;
 
   /// Whether to require the action from a specific environment.
-  bool? requireEnvironment;
+  bool requireEnvironment;
 
   /// The GitHub environment where the publishing is required from.
   String? environment;
 
+  /// Whether the (default) `push` event is enabled.
+  bool isPushEventEnabled;
+
+  /// Whether the `workflow_dispatch` event is enabled.
+  bool isWorkflowDispatchEventEnabled;
+
   GithubPublishingConfig({
-    this.isEnabled,
+    this.isEnabled = false,
     this.repository,
     this.tagPattern,
-    this.requireEnvironment,
+    this.requireEnvironment = false,
     this.environment,
+    this.isPushEventEnabled = true,
+    this.isWorkflowDispatchEventEnabled = false,
```

Comment:
```suggestion
    // MUST default to true, because this was the default behavior before
    // isWorkflowDispatchEventEnabled was introduced!
    this.isPushEventEnabled = true,
    // MUST default to false, because the default behavior prior to the
    // introduction of isWorkflowDispatchEventEnabled was to only allow
    // publishing from push events.
    this.isWorkflowDispatchEventEnabled = false,
```